Starting database

Keparlg opened the pull request:
Added DBHelper, ChannelsRepository, PostsRepository, TeamsRepository, UsersRepository. 
To DBHelper added creation for channels, posts, teams and users tables. 
Added MattermostContentProvider (empty class yet).
Renamed table fields constants names.
Added constants for uries and codes to MattermostContentProvider.
Made major modifications at query() method of MattermostContentProvider

Keparlg committed:
Added DBHelper, ChannelsRepository, PostsRepository, TeamsRepository, UsersRepository. To DBHelper added creation for channels, posts, teams and users tables. Added MattermostContentProvider (empty class yet)

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/MattermostContentProvider.java:
```diff
@@ -0,0 +1,49 @@
+ */
+
+public class MattermostContentProvider extends ContentProvider {
+    static final String DB_NAME = "mattermostDb";
```
на мой взгляд на конец файла нужно поставить, расширение .db

kraftuK replied:
и желательно название файла БД и версию вынести внутрь sqliteHelper

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/DBHelper.java:
```diff
@@ -0,0 +1,95 @@
+ * Created by kepar on 06.02.17.
+ */
+
+public class DBHelper extends SQLiteOpenHelper {
```
По умолчанию поддержку вторичных ключей нужно включать руками 
    @Override
    public void onConfigure(SQLiteDatabase db) {
        db.setForeignKeyConstraintsEnabled(true);
    }

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/repository/ChannelsRepository.java:
```diff
@@ -0,0 +1,22 @@
+ * Created by kepar on 06.02.17.
+ */
+
+public class ChannelsRepository {
```
Этот класс должен implements BaseColumns
Поля  FIELD_NAME_TYPE, мне кажется нужно уменьшить к примеру брать FIELD_NAME, очень длинные строки получаются, пример:
ChannelsRepository.FIELD_NAME_HEADER
или
ChannelsRep.HEADER

Keparlg replied:
Я считаю, что более короткие названия способствуют ухудшению читаемости кода

kraftuK replied:
да я с тобой полностью согласен. Все зависит что ты хочешь в этом классе делать.
Просто название класса, долго говорить за себя, для чего он нужно, пример:
1) Название класса ChannelsRepository и у него поля FIELD_NAME_HEADER
В итоге мы используем ChannelsRepository.FIELD_NAME_HEADER
2) название класса ChannelsColumns implements BaseColumns и просто поля  вида HEADER
В итоге мы используем ChannelsColumns.HEADER
С одной стороны не критично, оставляем как есть.

Keparlg replied:
Ну так этот класс должен хранить в себе не только названия полей. Почему он должен называться XxxxColumns?

kraftuK reviewed (commented):
Вова, посмотри комментария к файлам.

kraftuK commented:
Вот тут очень хорошо написано как делать надо:
https://git.kilograpp.com/android/test_certificate

Keparlg committed:
Renamed table fields constants names.
Added constants for uries and codes to MattermostContentProvider.
Made major modifications at query() method of MattermostContentProvider

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/DBHelper.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.setForeignKeyConstraintsEnabled(true);
```
Вова, надо переопределить метод, так вроде нельзя.

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/DBHelper.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
```
Тут надо удалить все таблицы и потом вызвать метод onCreate, хоть и нет пока что миграции.

kraftuK commented on app/src/main/java/com/kilogramm/mattermost/database/MattermostContentProvider.java:
```diff
@@ -0,0 +1,139 @@
+            case CODE_CHANNELS_WITH_ID:
+                tableName = ChannelsRepository.TABLE_NAME_CHANNELS;
+                notificationUri = CONTENT_URI_CHANNELS;
+                break;
```
в чем разница логика работы CODE_CHANNELS и CODE_CHANNELS_WITH_ID ? 
Если я не ошибаюсь, тут должно в selection, добавляться параметр нахождения конкретной записи в таблицы, если CODE_CHANNELS_WITH_ID.

kraftuK reviewed (approved):
В целом норм.

Keparlg committed:
Added FileInfosRepository
Changed Foreign keys to right
AddedPropsRepository
AddedAttachmentsRepository

Keparlg committed:
Common ContentProvider changes
Changed name of database;
Changed common table's id declaration
Added MattermostContentProvider.insert() method realisation
Added MattermostContentProvider.delete() method realisation
Added MattermostContentProvider.update() method realisation

Keparlg committed:
Change MattermostContentProvider.delete() method

Keparlg committed:
Add exported=false property for provider
